python/pyspark/logger/tests/test_logger.py

```diff
@@ -0,0 +1,128 @@
+        log_json = json.loads(self.handler.stream.getvalue().strip())
+        keys = ["ts", "level", "logger", "msg", "context"]
+        for key in keys:
+            self.assertTrue(key in log_json)
+
+    def test_log_info(self):
+        self.logger.info("This is an info log", user="test_user_info", action="test_action_info")
+        log_json = json.loads(self.handler.stream.getvalue().strip())
+
+        self.assertEqual(log_json["msg"], "This is an info log")
+        self.assertEqual(
+            log_json["context"], {"action": "test_action_info", "user": "test_user_info"}
+        )
+        self.assertTrue("exception" not in log_json)
+
+    def test_log_warn(self):
+        self.logger.warn("This is an warn log", user="test_user_warn", action="test_action_warn")
+        log_json = json.loads(self.handler.stream.getvalue().strip())
+
+        self.assertEqual(log_json["msg"], "This is an warn log")
+        self.assertEqual(
+            log_json["context"], {"action": "test_action_warn", "user": "test_user_warn"}
+        )
+        self.assertTrue("exception" not in log_json)
+
+    def test_log_error(self):
+        self.logger.error(
+            "This is an error log", user="test_user_error", action="test_action_error"
+        )
+        log_json = json.loads(self.handler.stream.getvalue().strip())
+
+        self.assertEqual(log_json["msg"], "This is an error log")
+        self.assertEqual(
+            log_json["context"], {"action": "test_action_error", "user": "test_user_error"}
+        )
+        self.assertTrue("exception" not in log_json)
+
+    def test_log_exception(self):
+        self.logger.exception(
+            "This is an exception log", user="test_user_exception", action="test_action_exception"
+        )
+        log_json = json.loads(self.handler.stream.getvalue().strip())
+
+        self.assertEqual(log_json["msg"], "This is an exception log")
+        self.assertEqual(
+            log_json["context"], {"action": "test_action_exception", "user": "test_user_exception"}
+        )
+        self.assertTrue("exception" in log_json)
+        self.assertTrue("class" in log_json["exception"])
+        self.assertTrue("msg" in log_json["exception"])
+        self.assertTrue("stacktrace" in log_json["exception"])
+
+    def test_dataframe_query_context_logger(self):
+        with self.sql_conf({"spark.sql.ansi.enabled": True}):
+            df = self.spark.range(10)
+
+            with self.assertRaises(ArithmeticException):
+                df.withColumn("div_zero", df.id / 0).collect()
+            log_json = json.loads(self.handler.stream.getvalue().strip())
+
+            err_msg = "[DIVIDE_BY_ZERO] Division by zero."
+            self.assertTrue(err_msg in log_json["msg"])
+            self.assertTrue(err_msg in log_json["exception"]["msg"])
+            self.assertEqual(log_json["context"]["fragment"], "__truediv__")
+            self.assertEqual(log_json["context"]["error_class"], "DIVIDE_BY_ZERO")
```

Comment:
I think `message_parameters` and `error_class` has to be camel cased. Seems like it's preexisting issue so you won't have to deal with it here but I think we should probably fix them

Reply:
Got it. Let me fix them within a separate ticket after completing this PR